gatsby-node.js

```diff
@@ -99,4 +106,253 @@ async function makeSynoptic(createPage, reporter, graphql) {
       context: section
     })
   }
+}
+
+async function makeSearchIndex(reporter, graphql){
+  const lunr = require('lunr')
+  const remark = (await import('remark')).remark
+
+  const jsdom = require("jsdom")
+  const { JSDOM } = jsdom
+
+  const result_md = await graphql(`
+    query {
+      allMarkdownRemark {
+        edges {
+          node {
+            frontmatter {
+              path
+              title
+            }
+            rawMarkdownBody
+          }
+        }
+      }
+    }
+  `)
+  const result_synoptic = await graphql(`
+    query {
+      allCetei {
+        nodes {
+          prefixed
+          elements
+          parent {
+            ... on File {
+              name
+              relativeDirectory
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  if (result_md.errors || result_synoptic.errors) {
+    reporter.panicOnBuild(`Error while running GraphQL query.`)
+    return
+  }
+
+
+  const searchIndex = lunr(function () {
+    this.ref('path')
+    this.field('title')
+    this.field('heading')
+    this.field('content')
+    this.metadataWhitelist = ['position']
+  
+    result_md.data.allMarkdownRemark.edges.forEach(({ node }) => {
+      const tree = remark().parse(node.rawMarkdownBody)
+      let heading = ""
+      let content = ""
+
+      for(let i = 0; i < tree.children.length; i++){
```

Comment:
Change this to a for .. of. It's more readable. Unless `tree.children` is not iterable? You could still consider using `Array.from(tree.children)`, or using `tree.children.forEach()` 

```js
for (const child of tree.children) {
   // ...
  
  if (child.type === 'heading') {}
  
   // ...  
}
```

Reply:
If these are NodeLists, consider converting them to Arrays with Array.from. It will simplify things greatly. Some more info here: https://stackoverflow.com/questions/24775725/loop-through-childnodes#comment69532209_24775765